src/cpp/session/modules/SessionDataImportV2.R

```diff
@@ -13,21 +13,162 @@
 #
 #
 
-.rs.addJsonRpcHandler("preview_data_import", function(path, maxRows, maxCols = 100, maxFactors = 64)
+.rs.addFunction("assemble_data_import_name", function(dataImportOptions)
+{
+  dataName <- dataImportOptions$dataName
+
+  if (is.null(dataName) || dataName == "")
+    {
+      if (!is.null(dataImportOptions$importLocation) && dataImportOptions$importLocation != "")
+      {
+        basename <- basename(dataImportOptions$importLocation)
+        if (length(basename) > 0)
+        {
+          fileComponents <- unlist(strsplit(basename, ".", fixed = TRUE))
+          components <- length(fileComponents)
+          if (components >= 1)
+          {
+            dataName <- paste(fileComponents[1:components-1])
+          }
+        }
+      }
+    }
+  
+  dataName
+})
+
+.rs.addFunction("assemble_data_import_parameters_csv", function(dataImportOptions)
+{
+  parameters <- ""
+  spacing <- ",\n  "
+  useCsv <- (!is.null(dataImportOptions$delimiter) && dataImportOptions$delimiter == ",")
```

Comment:
You may have already discovered that `NULL == "foo"` results in a zero-length logical vector (`logical(0)`) rather than `FALSE`. :-) There's a more succinct way of doing this test:

`identical(dataImportOptions$delimiter, ",")`

This works because `identical` always returns a single-length logical vector (i.e. `identical(NULL, ",")` is `FALSE`).